dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/teradata/TeradataLogsJdbcTask.java

```diff
@@ -190,7 +190,12 @@ private String getSql(@Nonnull JdbcHandle handle) {
       // "QueryID is a system-wide unique field; you can use QueryID
       // to join DBQL tables ... without needing ProcID as an additional join field."
       // (https://docs.teradata.com/reader/B7Lgdw6r3719WUyiCSJcgw/YIKoBz~QQgv2Aw5dF339kA)
-      buf.append(" LEFT OUTER JOIN ").append(queryTable).append(" ST ON L.QueryID=ST.QueryID");
+      buf.append(" LEFT OUTER JOIN ").append(queryTable).append(" ST ON (L.QueryID=ST.QueryID");
+
+      if (logDateColumn != null) {
+        buf.append(" AND L.").append(logDateColumn).append("=ST.").append(logDateColumn);
```

Comment:
Discussed offline the interaction between this and the condition at line 216 which could make the join an inner join.

Reply:
I've removed the condition on `ST.LogDate` column from the `WHERE` clause, but as suggested by another person, this will not make a difference, because it will still be treated as an inner join.